sky/backends/cloud_vm_ray_backend.py

```diff
@@ -4102,11 +4130,16 @@ def _execute_file_mounts(self, handle: CloudVmRayResourceHandle,
         start = time.time()
         ip_list = handle.external_ips()
         port_list = handle.external_ssh_ports()
+        run_on_k8s = isinstance(handle.launched_resources.cloud,
+                                clouds.Kubernetes)
         assert ip_list is not None, 'external_ips is not cached in handle'
         ssh_credentials = backend_utils.ssh_credential_from_yaml(
             handle.cluster_yaml, handle.docker_user)
         runners = command_runner.SSHCommandRunner.make_runner_list(
-            ip_list, port_list=port_list, **ssh_credentials)
+            ip_list,
+            port_list=port_list,
+            run_on_k8s=run_on_k8s,
```

Comment:
Instead of explicitly specifying the `run_on_k8s`, it might be better to have this argument generated by `ssh_credentials` argument, where we can check the cloud provider in the YAML file?

Reply:
nit for the naming: maybe we can call it `proxy_to_k8s`?

Reply:
@Michaelvll Thanks for the direction! This is updated now.